### PR TITLE
Enhancement/change file filtering logic

### PR DESCRIPTION
**Pull request by tristanhyams**
## Description
#5089 Changes the file filtering logic for match-query to AND instead of OR. This means if multiple queries are searched for within the filter field, only entries that match all queries will be displayed.

## Changes

- Added check to see whether item matches all queries. If all queries do not match, match-query will return false. 
- Updated CHANGELOG.md

## Additional information
yarn test passes and yarn lint reports no errors

Tested on: MacOs 14.3.1

### CODE REVIEW

**Comment by nathanlesage** [comment]
Thank you. We'll have to solve the merge conflicts with the Changelog, but in general I appreciate that you took the time to write a few lines indicating this change for users.

@kyaso Do you have time to review this PR before your vacation? It doesn't have urgent priority, but I'm booked until the end of June, and so I want to save my resources for potential urgent fixes.

**Review by kyaso — APPROVED** [review]
LGTM. Thanks for the PR!

File: source/win-main/file-manager/util/match-query.ts [hunk 1]
```diff
@@ -32,48 +32,62 @@ import type { AnyDescriptor } from '@dts/common/fsal'
 export default function matchQuery (query: string, includeTitle: boolean, includeH1: boolean): (item: AnyDescriptor) => boolean {
   const queries = query.split(' ').map(q => q.trim()).filter(q => q !== '')
 
-  // Returns a function that takes a Meta descriptor and returns whether it matches or not
+  // Returns a function that takes a Meta descriptor and returns whether it matches all queries or not
   return function (item: AnyDescriptor): boolean {
+    // Initialize a variable to keep track of whether all queries are matched
```
Comment (nathanlesage) [inline source/win-main/file-manager/util/match-query.ts:37]:
```suggestion
```

File: source/win-main/file-manager/util/match-query.ts [hunk 2]
```diff
@@ -32,48 +32,62 @@ import type { AnyDescriptor } from '@dts/common/fsal'
...
+    let allQueriesMatched = true
+
     for (const q of queries) {
+      let queryMatched = false // Track if the current query is matched
```
Comment (nathanlesage) [inline source/win-main/file-manager/util/match-query.ts:41]:
```suggestion
      let queryMatched = false
```

File: source/win-main/file-manager/util/match-query.ts [hunk 3]
```diff
@@ -32,48 +32,62 @@ import type { AnyDescriptor } from '@dts/common/fsal'
...
+        queryMatched = true
       }
 
-      if (item.type !== 'file') {
-        continue // The rest can only match files
-      }
+      // The rest can only match files
+      if (item.type === 'file') {
+        // Type assertion to check if 'firstHeading' exists on file descriptors
+        const fileDescriptor = item as { type: 'file', tags: string[], frontmatter?: Record<string, any>, firstHeading?: string | null }
```
Comment (nathanlesage) [inline source/win-main/file-manager/util/match-query.ts:51]: Can you directly coerce to MDFileDescriptor?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+efficient way to search for files, as often multiple queries will want to be
+searched, and all files matching all queries will be displayed.
+
+In this latest update, Zettlr has changed to AND file filtering logic, meaning
+that only items that match all queries will be displayed when searching with
+multiple queries.
+
+## GUI and Functionality
+
+(Nothing here)
```
Comment (nathanlesage) [inline CHANGELOG.md:17]: Please indicate this using **Change**: short description

(See previous changelog for how)

**Review by nathanlesage — APPROVED** [review]
Thanks @kyaso for the code review, I just have a few "formalities" regarding the changelog :)

**Comment by tristanhyams** [comment]
Thanks for the help @kyaso @nathanlesage. Have implemented these changes :)

**Comment by nathanlesage** [comment]
Looks good, thank you! Unfortunately I'm on mobile right now and the app indicates that there are merge conflicts. I have little time to manually solve them over the weekend, but should you have time, I can merge in a calm minute so that we can mark this PR as done.

**Comment by tristanhyams** [comment]
> Looks good, thank you! Unfortunately I'm on mobile right now and the app indicates that there are merge conflicts. I have little time to manually solve them over the weekend, but should you have time, I can merge in a calm minute so that we can mark this PR as done.

All done, should be good to merge

**Comment by nathanlesage** [comment]
Perfect! Thank you for reacting so quickly!

**Pull request merged**